truffle-config.js

```diff
@@ -1,42 +1,54 @@
 const HDWalletProvider = require('@truffle/hdwallet-provider');
 
 const fs = require('fs');
-const mnemonic = fs.readFileSync(".secret").toString().trim();
-if (!mnemonic || mnemonic.split(' ').length !== 12) {
-  throw new Error('unable to retrieve mnemonic from .secret');
+let mnemonic;
+try {
+  mnemonic = fs.readFileSync(".secret").toString().trim();
+} catch (e) {
+  mnemonic = 'INVALID';
 }
 
-const gasPriceTestnetRaw = fs.readFileSync(".gas-price-testnet.json").toString().trim();
-const gasPriceTestnet = parseInt(JSON.parse(gasPriceTestnetRaw).result, 16);
-if (typeof gasPriceTestnet !== 'number' || isNaN(gasPriceTestnet)) {
-  throw new Error('unable to retrieve network gas price from .gas-price-testnet.json');
-}
-console.log("Gas price Testnet: " + gasPriceTestnet);
-
 module.exports = {
-
   mocha: {
     enableTimeouts: false,
     timeout: 1000000
   },
-
+  plugins: ["solidity-coverage"],
   networks: {
-    development: {
-         host: "127.0.0.1",     // Localhost (default: none)
-         port: 7777,            // Standard Ethereum port (default: none)
-         network_id: "*",       // Any network (default: none)
-        },
-
-    testnet: {
-      provider: () => new HDWalletProvider(mnemonic, 'http://localhost:7777'),
-      network_id: "*",
-      gasPrice: Math.floor(gasPriceTestnet * 1.1),
-      networkCheckTimeout: 1e9
+    rskRegtest: {
+      host: '127.0.0.1',
+      port: 4444,
+      network_id: 33,
+    },
+    rskTestnet: {
+      provider: () => new HDWalletProvider({
+        mnemonic,
+        providerOrUrl: `https://public-node.testnet.rsk.co`,
+        derivationPath: "m/44'/37310'/0'/0/"
+      }),
+      network_id: 31,
+      gasPrice: 6000000000,
     },
+    rskMainnet: {
+      provider: () => new HDWalletProvider({
+        mnemonic,
+        providerOrUrl: `https://public-node.rsk.co`,
```

Comment:
Not sure it is a good idea to use a public node here. I was having a lot of connectivity issues when I tried to run the tests against it.